File: java/src/main/java/com/azure/data/cosmos/serialization/hybridrow/schemas/PrimarySortKey.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.data.cosmos.serialization.hybridrow.schemas;

/**
 * Describes a property or set of properties used to order the data set within a single
 * partition.
 */
public class PrimarySortKey {

    private SortDirection direction = SortDirection.values()[0];
    private String path;

    /**
     * The logical path of the referenced property.
     * Primary keys MUST refer to properties defined within the same {@link Schema}.
     */
    public final SortDirection direction() {
        return this.direction;
    }

    public final PrimarySortKey direction(SortDirection value) {
        this.direction = value;
        return this;
    }

    /**
     * The logical path of the referenced property.
     * Primary keys MUST refer to properties defined within the same {@link Schema}.
     */
    public final String path() {
        return this.path;
    }

    public final PrimarySortKey path(String value) {
        this.path = value;
        return this;
    }
}

File: src/Serialization/HybridRow/Schemas/PrimarySortKey.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
// ------------------------------------------------------------

namespace Microsoft.Azure.Cosmos.Serialization.HybridRow.Schemas
{
    using Newtonsoft.Json;

    /// <summary>
    /// Describes a property or set of properties used to order the data set within a single
    /// partition.
    /// </summary>
    public sealed class PrimarySortKey
    {
        /// <summary>The logical path of the referenced property.</summary>
        /// <remarks>Primary keys MUST refer to properties defined within the same <see cref="Schema" />.</remarks>
        [JsonProperty(PropertyName = "path", Required = Required.Always)]
        public string Path { get; set; }

        /// <summary>The logical path of the referenced property.</summary>
        /// <remarks>Primary keys MUST refer to properties defined within the same <see cref="Schema" />.</remarks>
        [JsonProperty(PropertyName = "direction", Required = Required.DisallowNull)]
        public SortDirection Direction { get; set; }
    }
}
